### PR TITLE
yolo_master_df

### PR DESCRIPTION
**Pull request by ALLUPRASAD**

### CODE REVIEW

**Review by satheeshkatipomu — CHANGES_REQUESTED** [review]
1. Please resolve `flake8` formatting issues in your code, check the workflow log for more details about errors.
2. If you are using any new package which is not already present. Please add that package as a dependency.
3. Please check why docs build is failing.
4. Please add a short description of what is changed/implemented while raising in PR.
5. Please use a more descriptive commit message instead of generic messages.
6. Please change your `git config` before committing, so that you have correct details in the `git`  commit log.

**Comment by bishwarup307** [comment]
@ALLUPRASAD let's get rid of the `num2words` dependency and just encode the digits as strings if there is no `yaml` or `txt` file provided by the user.

**Comment by bishwarup307** [comment]
@ALLUPRASAD also the `FormatSpec` and the `Annotation` class need to be modified to account for the `yolo` format.

**Comment by ALLUPRASAD** [comment]
@bishwarup307 ,The changes required in `base.py `and `__init__.py` are already there for yolo.I think i don't need to change anything 
@satheeshkatipomu ,I have made requested changes please review

**Comment by bishwarup307** [comment]
Yes, I missed that.

File: optical/converter/yolo.py [hunk 1]
```diff
@@ -1,17 +1,136 @@
 """
-__author__: HashTagML
+__author__:HashTagML
```
Comment (satheeshkatipomu) [inline optical/converter/yolo.py:2]: Please format this

File: optical/converter/yolo.py [hunk 2]
```diff
@@ -1,17 +1,136 @@
...
 license: MIT
 Created: Wednesday, 31st March 2021
 """
 
-from .base import FormatSpec
```
Comment (satheeshkatipomu) [inline optical/converter/yolo.py:7]: Please follow this:
there is a pattern of organizing your imports. it goes like this:

=========

import packages that comes with base python e.g., os, pathlib, json etc.

import third-party packages e.g. numpy, pandas, tqdm etc.

import modules from the projects e.g. the relative imports

=========
There is a command in vscode in the command palette called "organize imports" which does that automatically for you.

File: optical/converter/yolo.py [hunk 3]
```diff
@@ -1,17 +1,136 @@
...
+import pandas as pd
+from .base import FormatSpec
+from .utils import exists, get_image_dir, get_annotation_dir
 
 
 class Yolo(FormatSpec):
     def __init__(self, root: Union[str, os.PathLike]):
-        pass
+        self.root = root
+        self.class_file = Path(os.path.join(self.root, "data.yaml"))
```
Comment (satheeshkatipomu) [inline optical/converter/yolo.py:23]: I think it is better if a user provides `yaml` file name and path. If not warn the user that you are by default searching for `data.yaml`

File: optical/converter/yolo.py [hunk 4]
```diff
@@ -1,17 +1,136 @@
...
+                    box_heights = splited[4]
+                    img_names.append(file_names)
+                    x_center.append(x_cent)
+                    y_center.append(y_cent)
+                    box_width.append(box_widths)
+                    box_height.append(box_heights)
+                    cls_ids.append(class_id)
+                    splits.append(split)
+
+        if self.class_file.is_file():
```
Comment (satheeshkatipomu) [inline optical/converter/yolo.py:81]: Better to move this snippet to a separate method, so that it is modularized.

Comment (bishwarup307) [reply optical/converter/yolo.py:81]: also more concise:
```
class_id, x_cent, y_cent, ... = first_line.split()
```

**Review by bishwarup307 — CHANGES_REQUESTED** [review]

**Comment by bishwarup307** [comment]
@ALLUPRASAD please avoid using same commit message for multiple commits.

**Review by bishwarup307 — APPROVED** [review]

**Review by satheeshkatipomu — APPROVED** [review]

**Pull request merged**